src/calibrationlib/calibration_lqp_generator.cpp

```diff
@@ -0,0 +1,163 @@
+#include "calibration_lqp_generator.hpp"
+#include <expression/expression_functional.hpp>
+#include <logical_query_plan/join_node.hpp>
+#include "logical_query_plan/predicate_node.hpp"
+
+#include <synthetic_table_generator.hpp>
+#include "storage/table.hpp"
+
+using opossum::expression_functional::between_inclusive_;
+using opossum::expression_functional::equals_;
+using opossum::expression_functional::greater_than_;
+using opossum::expression_functional::is_not_null_;
+using opossum::expression_functional::is_null_;
+using opossum::expression_functional::less_than_;
+using opossum::expression_functional::like_;
+using opossum::expression_functional::not_in_;
+
+namespace opossum {
+void CalibrationLQPGenerator::generate(OperatorType operator_type,
+                                       const std::shared_ptr<const CalibrationTableWrapper>& table) {
+  switch (operator_type) {
+    case OperatorType::TableScan:
+      _generate_table_scans(table);
+      return;
+    default:
+      break;
+  }
+  throw std::runtime_error("Not implemented yet: Only TableScans and JoinHashes are currently supported");
+}
+
+void CalibrationLQPGenerator::_generate_table_scans(const std::shared_ptr<const CalibrationTableWrapper>& table) {
```

Comment:
table > table_wrapper?
You access the actual table by `table->get_table()`. Confusing. Maybe `table_wrapper->get_table()`?